### PR TITLE
[ci] convert sharding from bash to python

### LINKED ISSUE #1905: label tutorials as multigpu vs not
Sometimes tutorials CI breaks because tutorials get sharded to the wrong machine so we should label tutorials that require multigpu machine vs. not

cc @malfet @janeyx99

### PR DESCRIPTION
**Pull request by clee2000**
fixes #1905
helps w/ #1899
Changes:
- convert sharding from bash to python
- create a metadata file to list long running jobs + jobs that require special machines
- shard based on time
  - default to 1 min if not a super long time
- only 0th worker uses gpu.nvidia.small.multi

There are three files (can see them in the metadata file) that are long running (20-30 minutes) and are responsible for the unbalance in the shards.  The long run time comes from running a training loop, so we should see what can be done about not running this training loop everytime (perhaps caching the result or reducing the number of epochs)

### CODE REVIEW

File: .jenkins/get_files_to_run.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+def main():
+    num_shards = int(os.environ.get("NUM_WORKERS", 20))
+    shard_num = int(os.environ.get("WORKER_ID", 0))
+
+    all_files = get_all_files()
+    files_to_run = calculate_shards(all_files, num_shards=num_shards)[shard_num]
+    remove_other_files(all_files, files_to_run)
+    stripped_file_names = list(map(lambda x: Path(x).stem, files_to_run))
+    print(" ".join(stripped_file_names))
```
Comment (janeyx99) [inline .jenkins/get_files_to_run.py:75]: Could these be separated by new lines for better readability in the logs?

**Review by janeyx99 — APPROVED** [review]

File: .jenkins/get_files_to_run.py (hunk 1 above)
Comment (clee2000) [reply .jenkins/get_files_to_run.py]: I added a newline so if you run the python script on its own there are new lines, but the output gets swallowed up by FILES_TO_RUN and printing that doesn't seem to include the new line unfortunately

File: .circleci/regenerate.py [hunk 2]
```diff
@@ -25,8 +25,8 @@ def indent(indentation, data_list):
 
 def jobs(pr_or_master, num_workers=20, indentation=2):
     jobs = {}
-    needs_gpu_nvidia_small_multi = [3, 9, 12, 13]
-    needs_gpu_nvidia_medium = [14, 15]
+    needs_gpu_nvidia_small_multi = [0]
+    needs_gpu_nvidia_medium = []
```
[the remark this answers is not in the document]
Comment (clee2000) [reply .circleci/regenerate.py]: removed
